### PR TITLE
Idan/fix/ generators windows support

### PR DESCRIPTION
**Pull request by iYadgar**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: [CONTRIBUTING.md](CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: apps/cli-daemon/src/app/generators/generators.service.ts [hunk 1]
```diff
@@ -36,23 +38,13 @@ export class GeneratorsService {
   }
 
   getAllGenerators(): GeneratorDefinition[] {
-    const schemasPath = pathResolve(
-      this.sessionService.cwd,
-      'node_modules/@schematics/angular'
-    );
```
Comment (ronnetzer) [inline apps/cli-daemon/src/app/generators/generators.service.ts:42]: I think that if you'll use path.resolve like this it'll work for all platforms:
```typescript
const schemasPath = pathResolve(
       this.sessionService.cwd, 'node_modules', '@schematics', 'angular'
     );
```

Comment (iYadgar) [reply apps/cli-daemon/src/app/generators/generators.service.ts:42]: Awesome, updated

File: apps/cli-daemon/src/app/generators/utils.ts [hunk 2]
```diff
@@ -10,29 +10,25 @@ const formatSchemaName = (name: string): string => {
     .map((name) => capitalize(name))
     .join('');
 };
-export const getGeneratorDefinition = (
-  generatorName: string,
+export const formatJsonToJs = <T>(path: string): T =>
+  JSON.parse(readFileSync(path, 'utf-8'));
```
Comment (ronnetzer) [inline apps/cli-daemon/src/app/generators/utils.ts:14]: Wrap in try-catch and throw meaningful errors

File: apps/cli-daemon/src/app/generators/utils.ts [hunk 3]
```diff
@@ -10,29 +10,25 @@ const formatSchemaName = (name: string): string => {
     .map((name) => capitalize(name))
     .join('');
 };
-export const getGeneratorDefinition = (
-  generatorName: string,
+export const formatJsonToJs = <T>(path: string): T =>
```
Comment (ronnetzer) [inline apps/cli-daemon/src/app/generators/utils.ts:13]: IMO if it's a function export it as a function

File: apps/cli-daemon/src/app/generators/utils.ts [hunk 4]
```diff
@@ -10,29 +10,25 @@ const formatSchemaName = (name: string): string => {
...
-  try {
-    const schema: Schema = JSON.parse(
-      readFileSync(`${path}/${generatorName}/schema.json`, 'utf-8')
-    );
-    const displayName = formatSchemaName(generatorName);
+): GeneratorDefinition[] => {
+  const collection = formatJsonToJs<{
+    schematics: Record<string, SchemaCollection>;
+  }>(path);
+  console.log('collection:', collection);
```
Comment (ronnetzer) [inline apps/cli-daemon/src/app/generators/utils.ts:21]: Remove log

File: apps/cli-daemon/src/app/generators/generators.service.ts [hunk 5]
```diff
@@ -61,4 +51,15 @@ export class GeneratorsService {
       return `${convertedKey}=${value}`;
     });
   }
+
+  private getPath(extension: string[]): string {
+    console.log('this.sessionService.cwd:', this.sessionService.cwd);
```
Comment (ronnetzer) [inline apps/cli-daemon/src/app/generators/generators.service.ts:56]: Remove log

File: apps/cli-daemon/src/app/generators/utils.ts [hunk 6]
```diff
@@ -10,29 +10,25 @@ const formatSchemaName = (name: string): string => {
     .map((name) => capitalize(name))
     .join('');
 };
-export const getGeneratorDefinition = (
-  generatorName: string,
+export const formatJsonToJs = <T>(path: string): T =>
+  JSON.parse(readFileSync(path, 'utf-8'));
+export const getGeneratorsDefinition = (
```
Comment (ronnetzer) [inline apps/cli-daemon/src/app/generators/utils.ts:15]: Consider adding try-catch here as well (not required if formatJsonToJs is already wrapped)

**Review by ronnetzer — APPROVED** [review]
LGTM

**Comment by agroupp** [comment]
Why is it all so much complicated? It must be simple thing:
```
import { readFile } from 'node:fs/promises';
import { join as pathJoin } from 'node:path';

const COLLECTION_FILE_PATH = pathJoin('node_modules', '@schematics', 'angular', 'collection.json');

try {
      const collection = JSON.parse(await readFile(pathJoin(this.sessionService.cwd, COLLECTION_FILE_PATH), { encoding: 'utf8' }));
      console.log(collection)
    } catch (err) {
      throw new NotFoundException('File collection.json not found or corrupted');
    }
```

**Pull request closed without merging**